packages/@ourworldindata/utils/src/Util.ts

```diff
@@ -1346,17 +1347,22 @@ export const imemo = <Type>(
 }
 
 export function recursivelyMapArticleContent<
-    Node extends OwidEnrichedGdocBlock | Span
+    Node extends OwidEnrichedGdocBlock | Span | EnrichedRecircLink
 >(
     node: Node,
-    callback: <Child extends OwidEnrichedGdocBlock | Span>(node: Child) => Child
+    callback: <Child extends OwidEnrichedGdocBlock | Span | EnrichedRecircLink>(
+        node: Child
+    ) => Child
 ): Node {
     if (checkNodeIsSpan(node)) {
         if ("children" in node) {
             node.children.map((node) =>
                 recursivelyMapArticleContent(node, callback)
             )
         }
+    } else if (node.type === "recirc") {
+        // slightly strange case where we need to iterate through recirc.links to extract URLs
+        node.links.map((link) => callback(link))
```

Comment:
nit: `forEach` feels more appropriate here since we're not using the result of `map`.

nit: rename to `recursivelyTraverseArticleContent`?

This is tangential to this change but I'm not quite getting why the callback or the function are returning anything at all, given we do nothing with the return values when we extract filenames or links?